docs/kubernetes/ingress-controller.md

```diff
@@ -163,6 +163,23 @@ TLS handshake error to the endpoint.  Other reasons are future support
 of features like session affinity, different load balancer
 algorithms or distributed loadbalancing also known as service mesh.
 
+### Using services instead of endpoints
+
+While using endpoints is the preferred way of using skipper as an
+ingress controller as described in the section above, there might be
+edge cases that require the use of [Kubernetes
+Services](http://kubernetes.io/docs/user-guide/services) instead.
+
+An example of scenario where you might need to use services is when you rely
+on Istio networking features to connect multiple clusters, as the IPs of
+Kubernetes endpoints will not resolve in all cases.
+
+If you find yourself in this category, you can override the default behaviour
+by setting the `KubernetesForceService` flag to `true` in the `Skipper.Options` struct.
+This will cause skipper to create routes with `BackendType=eskip.NetworkBackend` instead
+of `BackendType=eskip.LBBackend` and use the following address format:
+`http://<service name>.<namespace>.svc.cluster.local:<port>`.
```

Comment:
Maybe a link to the https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#services could be useful here

Reply:
Also `cluster.local` is a default and might be different for some clusters. 

https://stackoverflow.com/questions/59559438/retrieve-the-full-name-of-a-service-in-kubernetes claims kubernetes would automatically set up the DNS search domains so maybe we could get away with `<service name>.<namespace>.svc` that would nto depend on cluster domain

@szuecs WDYT?

Reply:
Let me know what you prefer @AlexanderYastrebov & @szuecs :)

Reply:
Always use `<service name>.<namespace>.svc.cluster.local` the other depends too much on cluster internal DNS setup. You can also make sure that search domains are not propagated and honestly always use long names for generated stuff, it will always be less error prone.

Reply:
I undid the change, it's back to `<service name>.<namespace>.svc.cluster.local`